src/ripple/app/main/Main.cpp

```diff
@@ -621,7 +621,8 @@ int run (int argc, char** argv)
                 std::cerr << "WARNING: using deprecated rpc_port param.\n";
                 try
                 {
-                    res.first.at_port(vm["rpc_port"].as<std::uint16_t>());
+                    res.first = res.first.at_port(
```

Comment:
can we use/add [[no_discard]] on these functions to catch similar bugs in the future?

Reply:
Hugely supportive of attribute annotations; not sure if `[[nodiscard]]` will generate an error or warning in  C++14 mode.

Reply:
It'll be ignored until C++17...  [cppreference](https://en.cppreference.com/w/cpp/language/attributes)

Reply:
I'm using CMake + Ninja on my machine, and it warned for `[[nodiscard]]`. It is using `g++` (aliased as `/usr/bin/c++`) with `-std=gnu++14`.

Reply:
Excellent point.  The standard only says what a compiler is required to consume.  The compiler can warn about anything it wants.  In this case the compiler is probably warning about either `[[nodiscard]]` being ignored or that there's an unknown attribute.

Reply:
In my test, it warned about the `[[nodiscard]]` return value being discarded, i.e. it had already implemented the C++17 required behavior, even for C++14. It might only be because we're using `-Wunused-result`:

```
../src/ripple/app/main/Main.cpp:625:60: warning:
    ignoring return value of ‘beast::IP::Endpoint beast::IP::Endpoint::at_port(beast::IP::Port) const’,
    declared with attribute nodiscard [-Wunused-result]
```